internal/ingress/annotations/class/main.go

```diff
@@ -44,7 +47,76 @@ var (
 func IsValid(ing *extensions.Ingress) bool {
 	ingress, ok := ing.GetAnnotations()[IngressKey]
 	if !ok {
-		glog.V(3).Infof("annotation %v is not present in ingress %v/%v", IngressKey, ing.Namespace, ing.Name)
+		glog.V(3).Infof("annotation %v is not present in object %v/%v", IngressKey, ing.Namespace, ing.Name)
+	}
+
+	// we have 2 valid combinations
+	// 1 - ingress with default class | blank annotation on ingress
+	// 2 - ingress with specific class | same annotation on ingress
+	//
+	// and 2 invalid combinations
+	// 3 - ingress with default class | fixed annotation on ingress
+	// 4 - ingress with specific class | different annotation on ingress
+	if ingress == "" && IngressClass == DefaultClass {
+		return true
+	}
```

Comment:
Let's extract the functionality out into a function rather than repeating it multiple times?